### PR TITLE
Exercise Endpoint Permissions

### PR DESCRIPTION
**Pull request by Philipp-g**

### CODE REVIEW

File: muesli/models.py [hunk 1]
```diff
@@ -687,11 +687,29 @@ def get_user(self, data):
         return usr
 
 
+class AssistantSchema(Schema):
```
Comment (TheEbolaDoc) [inline muesli/models.py]: Wouldnt it be possible to use inheritance from `UserSchema` here?

File: muesli/web/context.py [hunk 2]
```diff
@@ -318,3 +320,27 @@ def __init__(self, request):
...
+        exercise_id = exercise_id.strip("/")
+        user_id = request.matchdict.get('user_id', None)
+        self.exercise = request.db.query(Exercise).get(exercise_id)
+        self.lecture = self.exercise.exam.lecture
+        tutorial = None
+        self.user = None
+        if user_id is not None:
+            user_id = user_id.strip("/")
+            self.user = request.db.query(User).get(user_id)
+            tutorial = request.db.query(LectureStudent).filter(and_(LectureStudent.student_id == self.user.id, LectureStudent.lecture == self.lecture)).one().tutorial
```
Comment (TheEbolaDoc) [inline muesli/web/context.py]: What if `tutorial` is `None`?

File: muesli/web/viewsUser.py [hunk 3]
```diff
@@ -574,19 +569,17 @@ def list_auth_keys(request):
 @view_config(route_name='remove_api_key', context=context.GeneralContext)
 def removeKey(request):
     code_id = int(request.matchdict['key_id'])
-    #  TODO check if api_key.user_id matches user_id??
     api_key = request.db.query(models.BearerToken).get(code_id)
```
Comment (TheEbolaDoc) [inline muesli/web/viewsUser.py]: What if `code_id` is invalid?

File: muesli/web/viewsUser.py [hunk 4]
```diff
@@ -574,19 +569,17 @@ def list_auth_keys(request):
...
-    if not api_key:
-        request.session.flash('API Key nicht gefunden', queue='errors')
-    else:
+    if api_key.user == request.user or request.user.is_admin:
         api_key.revoked = True
         request.db.add(api_key)
         request.db.commit()
         request.session.flash('API Key entfernt ', queue='messages')
+    else:
+        request.session.flash('API Key nicht gefunden', queue='errors')
```
Comment (TheEbolaDoc) [inline muesli/web/viewsUser.py]: Why doesnt the error message represent what the `if` statement checks?

File: muesli/web/context.py (hunk 2 above)
Comment (Philipp-g) [reply muesli/web/context.py]: Good point

File: muesli/web/viewsUser.py (hunk 4 above)
Comment (Philipp-g) [reply muesli/web/viewsUser.py]: A person without permission shouldn't be able to deduct the existence of api keys from errors

File: muesli/web/viewsUser.py (hunk 3 above)
Comment (Philipp-g) [reply muesli/web/viewsUser.py]: Good point. Fixed now.

File: muesli/models.py (hunk 1 above)
Comment (Philipp-g) [reply muesli/models.py]: Good point. Changed

**Pull request merged**